runners/spark/src/main/java/org/apache/beam/runners/spark/translation/DoFnFunction.java

```diff
@@ -61,8 +65,20 @@ public DoFnFunction(DoFn<InputT, OutputT> fn,
       Exception {
     ProcCtxt ctxt = new ProcCtxt(mFunction, mRuntimeContext, mSideInputs);
     ctxt.setup();
-    mFunction.startBundle(ctxt);
-    return ctxt.getOutputIterable(iter, mFunction);
+    try {
+      mFunction.setup();
+      mFunction.startBundle(ctxt);
+      return ctxt.getOutputIterable(iter, mFunction);
+    } catch (Exception e) {
+      try {
```

Comment:
Maybe add comments, that this catch block handles issues in setup or startbundle, but that post-execution or during-execution failures are handled (somewhere)